### PR TITLE
Logical Partition Key In Query Plan Retrieval

### PR DESCRIPTION
**Pull request by bchong95**
# Logical Partition Key In Query Plan Retrieval

## Description

This PR adds the logical partition key to the query plan retrieval path. This is necessary for enabling the OFFSET + LIMIT pushdown that is possible for queries with logical partition keys. 

This PR will need to wait for this backend change to go through:

https://msdata.visualstudio.com/_git/CosmosDB/pullrequest/217129?_a=overview

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Query/QueryPartitionProvider.cs [hunk 1]
```diff
@@ -182,6 +185,7 @@ public void Update(IDictionary<string, object> queryengineConfiguration)
                         requireFormattableOrderByQuery,
                         isContinuationExpected,
                         allowNonValueAggregateQuery,
+                        //hasLogicalPartitionKey, // TODO: uncomment this when new ServiceInterop is deployed.
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Query/QueryPartitionProvider.cs]: Please create a GitHub issues and track it for GA.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.OnePartition.cs [hunk 2]
```diff
@@ -2638,6 +2638,73 @@ public async Task TestQueryCrossPartitionTop()
                 "/" + partitionKey);
         }
 
+        private async Task TestQueryCrossPartitionTopHelper(CosmosContainer container, IEnumerable<Document> documents)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.OnePartition.cs]: I guess in a related tests PR this file is removed.

Comment (bchong95) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CrossPartitionQueryTests.OnePartition.cs]: The diff is just weird. I moved the method up and deleted below.

**Comment by bchong95** [comment]
This PR is already covered here:

https://github.com/Azure/azure-cosmos-dotnet-v3/pull/378

**Pull request closed without merging**